### ISSUE TITLE
MEGAsync fails to build

### ISSUE THREAD

**Issue by JonathanReeve** [NixOS/nixpkgs#81387]
## Issue description

When trying to install `megasync`, it builds from source, and somewhere along the line the build fails. Here's what I think is the relevant error: 

```
g++ -c -pipe -std=c++11 -Wextra -Wconversion -Wno-unused-parameter -g -O2 -Wall -W -D_REENTRANT -fPIC -DQT_DISABLE_DEPRECATED_BEFORE=0x000000 -DHAVE_LIBUV -DUSE_MEDIAINFO -DUNICODE -DHAVE_LIBRAW -DHAVE_FFMPEG -DUSE_QT -DMEGA_QT_LOGGING -DUSE_SQLITE -DUSE_CRYPTOPP -DENABLE_SYNC -DENABLE_CHAT -DNDEBUG -DUSE_DBUS -DQT_NO_CAST_FROM_ASCII -DQT_NO_CAST_TO_ASCII -DQT_NO_DEBUG -DQT_SVG_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_DBUS_LIB -DQT_CORE_LIB -I. -Igui -Igui/linux -I/build/source/src/MEGASync/mega/bindings/qt/../..//bindings/qt/3rdparty/include/libuv -Imega/include -Imega/bindings/qt -I/build/source/src/MEGASync/mega/bindings/qt/../..//bindings/qt/3rdparty/include -Imega/include/mega/posix -Icontrol -Iplatform -Igoogle_breakpad -Iqtlockedfile -I. -isystem /nix/store/37dvkz512k8s87cx6qr12ircapb5m6gl-qtsvg-5.12.7-dev/include/QtSvg -isystem /nix/store/wjkic7izq98ylyvqcmmni9apcgw70hmx-qtbase-5.12.7-dev/include/QtWidgets -isystem /nix/store/wjkic7izq98ylyvqcmmni9apcgw70hmx-qtbase-5.12.7-dev/include/QtGui -isystem /nix/store/wjkic7izq98ylyvqcmmni9apcgw70hmx-qtbase-5.12.7-dev/include/QtNetwork -isystem /nix/store/wjkic7izq98ylyvqcmmni9apcgw70hmx-qtbase-5.12.7-dev/include/QtDBus -isystem /nix/store/wjkic7izq98ylyvqcmmni9apcgw70hmx-qtbase-5.12.7-dev/include/QtCore -I. -I/nix/store/y9alnmdfxk1xh68nbcdwkqzk40gpkqgc-mesa-19.3.3-dev/include -I. -I/nix/store/wjkic7izq98ylyvqcmmni9apcgw70hmx-qtbase-5.12.7-dev/mkspecs/linux-g++ -o linux_libc_support.o google_breakpad/common/linux/linux_libc_support.cc
make[1]: *** [Makefile:6825: exception_handler.o] Error 1
make[1]: *** Waiting for unfinished jobs....
google_breakpad/common/linux/linux_libc_support.cc: In function 'void my_uitos(char*, uintmax_t, unsigned int)':
google_breakpad/common/linux/linux_libc_support.cc:120:29: warning: conversion from 'uintmax_t' {aka 'long unsigned int'} to 'char' may change value [-Wconversion]
  120 |     output[index - 1] = '0' + (i % 10);
      |                         ~~~~^~~~~~~~~~
make[1]: Leaving directory '/build/source/src/MEGASync'
make: *** [Makefile:47: sub-MEGASync-make_first] Error 2
builder for '/nix/store/9abp3rsbrm79q4l9aiccckagqg4vx1nl-megasync-4.2.5.0.drv' failed with exit code 2
error: build of '/nix/store/9abp3rsbrm79q4l9aiccckagqg4vx1nl-megasync-4.2.5.0.drv' failed
```

Is it trying to build QT from source? And is there a reason it has to build stuff, rather than get binaries from the Nixpkgs cache? I have no idea what's going on here.

Also, FWIW, I tried bumping the version as in the patch below, but no dice—that fails to build, too. 

```
diff --git a/pkgs/applications/misc/megasync/default.nix b/pkgs/applications/misc/megasync/default.nix
index 6e51e3cda76..261561d25fa 100644
--- a/pkgs/applications/misc/megasync/default.nix
+++ b/pkgs/applications/misc/megasync/default.nix
@@ -26,13 +26,13 @@
 
 mkDerivation rec {
   pname = "megasync";
-  version = "4.2.3.0";
+  version = "4.2.5.0";
 
   src = fetchFromGitHub {
     owner = "meganz";
     repo = "MEGAsync";
     rev = "v${version}_Linux";
-    sha256 = "0l4yfrxjb62vc9dnlzy8rjqi68ga1bys5x5rfzs40daw13yf1adv";
+    sha256 = "1zw7x8gpvzhnzyirs5ishjl5idzyyin4wdxa67d6gzfgvqi33n7w";
     fetchSubmodules = true;
   };

```

### Steps to reproduce

- Run `nix-env -iA megasync`
- Notice that the build fails

## Technical details

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

 - system: `"x86_64-linux"`
 - host os: `Linux 5.5.6, NixOS, 20.09pre215024.e97dfe73bba (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.3`
 - channels(jon): `"home-manager, home-manager-sway, nixpkgs-20.09pre214168.686274ea62c"`
 - channels(root): `"nixos-20.09pre215024.e97dfe73bba"`
 - nixpkgs: `/home/jon/.nix-defexpr/channels/nixpkgs`

**Comment by thefloweringash** [comment]
Upstream issue: https://github.com/meganz/MEGAsync/issues/366

**Comment by michojel** [comment]
Resolved by #81117. Can we close this?

**Comment by JonathanReeve** [comment]
Sure.
